R/functions_netCDF.R

```diff
@@ -1,333 +1,420 @@
-      attval = "No temporal dimensions ... fixed field")
+  if(!isGridded) {
+    if(inherits(loc, 'Spatial')) {
+      nloc <- dim(loc@coords)[1]
+    } 
+    if(inherits(loc, 'sf')) {
+      nloc <- dim(locations)[1]
+    } 
   }
 
-  # add coordinate system attributes
-  prj <- raster::crs(x)
-  if (!is.na(prj)) {
-    ncdf4::ncatt_put(nc, varid = "crs", attname = "proj4",
-      attval = as.character(prj))
+  # Note: xvals should be organized from west to east, yvals from north to south
+  if(!is.null(grid)) {
+    xvals <- raster::xFromCol(grid, seq_len(raster::ncol(grid)))
+    yvals <- raster::yFromRow(grid, seq_len(raster::nrow(grid)))
+    grid_halfres <- raster::res(grid) / 2
+  } else {
+    if(inherits(loc, 'Spatial')) {
+      xvals <- sort(unique(loc@coords[,1]))
+      yvals <- sort(unique(loc@coords[,2]), decreasing = TRUE)
+      if(isGridded) grid_halfres <- loc@grid@cellsize / 2
+    }
+    if(inherits(loc, 'sf')) {
+      xvals <- sort(unique(st_coordinates(locations)[,1]))
+      yvals <- sort(unique(st_coordinates(locations)[,2]))
+      
+      if(isGridded) {
+        loc <- as(loc, Class = 'Spatial')
+        sp::gridded(loc) = TRUE
+        grid_halfres <- loc@grid@cellsize / 2      }
+    }
   }
+  
+  # crs attributes setup & info ----------------------------------------------------------------
+  if (!missing(crs_attributes)) {
 
-
-  #--- add variables values
-  for (k in seq_len(nl)) {
-    # write values to variable
-    vals <- raster::getValues(if (nl > 1) raster::raster(x, layer = k) else x)
-
-    try(ncdf4::ncvar_put(nc, varid = var_names[k],
-      vals = matrix(vals, ncol = var_chunksizes[2]),
-      start = var_start, count = var_chunksizes))
-
-    # add variable attributes
-    for (natt in ns_att_vars) {
-      ncdf4::ncatt_put(nc, varid = var_names[k], attname = natt,
-        attval = var_attributes[[natt]][k])
+    if ("proj" %in% names(crs_attributes)) {
+      proj <- crs_attributes[["proj"]]
+      crs_attributes[["proj"]] <- NULL
+    } else {
+      stop('Need "proj" in crs_attributes')
     }
+    
+    ns_att_crs <- names(crs_attributes)
 
-    # Flush this step to the file so we dont lose it
-    # if there is a crash or other problem
-    ncdf4::nc_sync(nc)
   }
 
-  invisible(TRUE)
-}
-
-#' @describeIn create_netCDF_from_raster_with_variables Convert array where
-#'   variables are organized in the third dimension to a \var{netCDF} file
-#' @export
-create_netCDF_from_array_with_variables <- function(x, locations, grid,
-  time_bounds, var_attributes, global_attributes, file, force_v4 = TRUE,
-  overwrite = FALSE) {
-
-  stopifnot(requireNamespace("ncdf4"))
-  if (force_v4) {
-    # avoid "_FillValue" error in older versions of `raster` package
-    stopifnot(utils::packageVersion("raster") >= "2.9.1")
+  # Time dimension setup & info ----------------------------------------------------------------
+  if(has_T_timeAxis & !is.null(time_bounds)) {
```

Comment:
`if` expects a single value (see `_R_CHECK_LENGTH_1_CONDITION_`) --> change `&` to `&&`